code/server/src/core/builtins/vehicle.h

```diff
@@ -55,16 +65,27 @@ namespace MafiaMP::Scripting {
         }
 
         void ChangeRadioStation(int id) {
-            FW_SEND_SERVER_COMPONENT_GAME_RPC(Shared::RPC::TuneRadio, _ent, id);
+            auto carData = _ent.get_mut<Shared::Modules::VehicleSync::UpdateData>();
+            carData->radioId = id;
+            MafiaMP::Shared::RPC::VehicleSetProps msg{};
+            msg.radioId = id;
+            FW_SEND_SERVER_COMPONENT_GAME_RPC(Shared::RPC::VehicleSetProps, _ent, msg);
         }
 
         void TurnOffRadio() {
-            FW_SEND_SERVER_COMPONENT_GAME_RPC(Shared::RPC::TuneRadio, _ent, -1);
+            auto carData = _ent.get_mut<Shared::Modules::VehicleSync::UpdateData>();
+            carData->radioId = -1;
+            MafiaMP::Shared::RPC::VehicleSetProps msg {};
+            msg.radioId = -1;
+            FW_SEND_SERVER_COMPONENT_GAME_RPC(Shared::RPC::VehicleSetProps, _ent, msg);
         }
```

Comment:
I think we should have `SetRadioStation`, `GetRadioStation`, `SetRadioEnabled`, `GetRadioEnabled ` instead.
It means sync `radioId` and the radio state.

Reply:
When we turn radio off the variable is set to -1 so we lose reference of which radio station was played, so `SetRadioEnabled` won't be helpful unless toggling it on would always pick ID 0

Reply:
We save some little memory by reusing a single variable to keep both radio station number and its state, but I can introduce another one just to keep reference of station number in case we turn radio off.